infra/enrolment-records-store/src/test/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSourceTest.kt

```diff
@@ -0,0 +1,404 @@
+
+import android.content.ContentResolver
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import com.simprints.core.domain.face.FaceSample
+import com.simprints.core.domain.fingerprint.FingerprintSample
+import com.simprints.core.domain.fingerprint.IFingerIdentifier.*
+import com.simprints.core.tools.json.JsonHelper
+import com.simprints.core.tools.utils.EncodingUtils
+import com.simprints.infra.enrolment.records.store.commcare.CommCareIdentityDataSource.Companion.COLUMN_DATUM_ID
+import com.simprints.infra.enrolment.records.store.commcare.CommCareIdentityDataSource.Companion.COLUMN_VALUE
+import com.simprints.infra.enrolment.records.store.domain.models.FaceIdentity
+import com.simprints.infra.enrolment.records.store.domain.models.FingerprintIdentity
+import com.simprints.infra.enrolment.records.store.domain.models.SubjectQuery
+import com.simprints.infra.logging.Simber
+import io.mockk.MockKAnnotations
+import io.mockk.Runs
+import io.mockk.clearAllMocks
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import io.mockk.unmockkStatic
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertTrue
+import kotlinx.coroutines.test.runTest
+import org.junit.AfterClass
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Test
+
+class CommCareIdentityDataSourceTest {
+
+    companion object {
+        private const val SUBJECT_ACTIONS_FINGERPRINT_1 =
+            "{\"events\":[{\"id\":\"0dafcd03-96c4-4ca5-b802-292da6d4f799\",\"payload\":{\"subjectId\":\"b26c91bc-b307-4131-80c3-55090ba5dbf2\",\"projectId\":\"nXcj9neYhXP9rFp56uWk\",\"moduleId\":{\"value\":\"AWuA3H0WGtHI2uod+ePZ3yiWTt9etQ==\"},\"attendantId\":{\"value\":\"AdySMrjuy7uq0Dcxov3rUFIw66uXTFrKd0BnzSr9MYXl5maWEpyKQT8AUdcPuVHUWpOkO88=\"},\"biometricReferences\":[{\"id\":\"2b9b4991-29d7-3eee-ac02-191afaa0c1a2\",\"templates\":[{\"quality\":99,\"template\":\"Rk1SACAyMAAAAADkAAABLAGQAMUAxQEAABBjIYCzAAgQAEAhABoeAIB3ACQNAEAoACsYAEDvADYDAECrAFgDAEBQAHIaAEDUAH2AAEEEAIX4AEDhAJX4AEDpALZ7AIB1AL0RAEEJAMPuAECSAM0NAED2ANPxAEDmAOXsAEDdAPrnAEBhAQYaAIDcARDaAICwARn8AEDJASlnAECMATcVAEB7ATgbAECxAT2qAECnAT6hAEB3AT6eAECIAT6eAEEWAUDGAIDHAUizAED9AU2zAEDsAVU9AED1AWYzAIC2AWciAAAA\",\"finger\":\"LEFT_THUMB\"},{\"quality\":88,\"template\":\"Rk1SACAyMAAAAAEUAAABLAGQAMUAxQEAABBYKUBqACiHAIBKAD0RAECyAEjuAEAfAF6sAEAnAG4sAIB4AHT2AIByAIuiAEAGAIs1AEC9AJDUAEB3AJS9AICaAJXRAEDzAJ/bAECHAKi9AIBzALJDAEAXALfDAEB6ALszAIDGAL3JAED2AL/QAEBHAMrRAEENAMrWAECiAMy9AIAkANfNAEDwANnKAEB3AOr4AICjAPKhAEANAPXGAIEEAPjJAEBCAPzaAICNAQUDAEEZAQdpAEDOARsfAIDpAR0TAEDZASUUAEDnASeNAIDsAS0NAED+ATONAEDxAT8KAIB9AUB5AEDAAVUNAEBdAVZzAECCAWt9AAAA\",\"finger\":\"LEFT_INDEX_FINGER\"}],\"format\":\"ISO_19794_2\",\"type\":\"FINGERPRINT_REFERENCE\"}]},\"type\":\"EnrolmentRecordCreation\"}]}"
```

Comment:
With long blobs of JSON it is much better to use the triple quote string as it does not require you to escape every quotemark and you can also just copy-paste the string contents.